Documentation/pages/docs/features/graphql-middleware.mdx

```diff
@@ -0,0 +1,84 @@
+import { Callout } from 'nextra-theme-docs'
+
+# GraphQL Middleware
+
+<Callout type="warning" emoji="🚧">
+**Work in progress** <br/>
+Resolver-based middleware is still a work in progress. 
+It is currently only supported as extensions to [Graphiti](https://github.com/GraphQLSwift/Graphiti), and only available in `v1.2.0-beta.1` or higher.
+</Callout>
+
+Middleware are useful reusable code that can be easily attached to resolvers. 
+By using middleware we can extract the commonly used code from our resolvers and then declaratively attach it whenever it's needed.
```

Comment:
Explain a bit more on why middleware is useful and how it differs from regular router middleware